### DISCUSSION TITLE
component.blade.php => component.x.php

### DISCUSSION THREAD

**Discussion by marknuyens** [laravel/framework#37220] [Ideas]
Hey there,

Are you tired of typing ".blade.php" over and over again?

Since we're mostly dealing with the "X" symbols when it comes to components in Blade, wouldn't it be nice to have support for an alias that allows for: `component.x.php` instead of `component.blade.php`.

Less characters, less typing, more doing. ;-)

Love to hear your opinions!

**Comment by s-patompong** [comment]
Good idea, been writing a lot of <x-...> tag lately.

**Comment by kohenkatz** [comment]
The `Illuminate\View\Factory` and `Illuminate\View\FileFinder` classes both have a method `addExtension` that you should be able to use to do this yourself (probably in the `boot` method of one of your `ServiceProvider`s).

The only thing you'll be missing is that the `artisan view:cache` command won't cache these files, so you will have to define your own version of that command if you want to pre-cache the views.
